src/module_PI/Raspberry_PI/main/PacketMessageTranslator.java

```diff
@@ -0,0 +1,110 @@
+		messageTable.put(5, "Execute Command: Trigger");
+		messageTable.put(6, "Execute Command: Reset Calibration");
+		messageTable.put(7, "Execute Command: Test Components");
+		messageTable.put(100, "Alarm object sucessfully created");
+		messageTable.put(101, "Alarm base photoresistor reading determined");
+		messageTable.put(102, "Alarm failed action");
+		messageTable.put(103, "Alarm successful action");
+		messageTable.put(104, "Alarm failed to arm");
+		messageTable.put(105, "Alarm sucessfully armed");
+		messageTable.put(106, "Alarm successful calibration");
+		messageTable.put(107, "Alarm has begun arming");
+		messageTable.put(108, "Alarm has been triggered");
+		messageTable.put(109, "Alarm calibration is being reset");
+		messageTable.put(110, "Alarm is being disarmed");
+		messageTable.put(111, "Alarm is being silenced");
+		messageTable.put(150, "Button object sucessfully created");
+		messageTable.put(151, "Button has been pressed");
+		messageTable.put(200, "Buzzer object has been created");
+		messageTable.put(201, "Buzzer has started to sound");
+		messageTable.put(202, "Buzzer has stopped sounding");
+		messageTable.put(250, "Component object sucessfully created");
+		messageTable.put(300, "ComponentTester default constructor called");
+		messageTable.put(301, "ComponentTester object sucessfully created");
+		messageTable.put(350, "Laser object sucessfully created");
+		messageTable.put(351, "Laser has been turned on");
+		messageTable.put(352, "Laser has been turned off");
+		messageTable.put(400, "LED object sucessfully created");
+		messageTable.put(401, "LED has been turned on");
+		messageTable.put(402, "LED has been turned off");
+		messageTable.put(403, "LED flashing");
+		messageTable.put(404, "LED toggling");
+		messageTable.put(450, "Photoresistor object has been successfully created");
+		messageTable.put(451, "Photoresistor taking reading");
+	}
+	
+	public String translate(int packetDetails){
+		if(isIntInScope(packetDetails)) throw new IllegalArgumentException("invalid packet int: " + packetDetails);
+		int correctedInt = parsePacketDetailsFromFullPacketInt(packetDetails);
+		if(!messageTable.containsKey(correctedInt)) throw new IllegalArgumentException("packetDetails " + packetDetails + " not in messageTable");
+		String packetMessage = messageTable.get(correctedInt);
+		return packetMessage;
+	}
+	
+	private int parsePacketDetailsFromFullPacketInt(int packetAsInt){
+		int packetModThisIsCorrectLength = (int) (Math.pow(10, NUM_OF_BYTES_IN_PACKET_DETAILS));
+		return packetAsInt % packetModThisIsCorrectLength;
+	}
+	
+	private boolean isIntInScope(int packetDetails){
+		return (packetDetails < 1 
+				|| packetDetails > MAX_PACKET_DETAILS_INT
+				|| packetDetails > MAX_DETAILS_ONLY_INT && packetDetails < MIN_FULL_PACKET_INT);
+	}
+	
+	public String translate(String packetDetailsAsString){
+		if(isStringNumeric(packetDetailsAsString)) return translate(Integer.parseInt(packetDetailsAsString));
```

Comment:
An alternative way to do this is just wrap the parseInt call with a try/catch block. Then you don't have to do the extra check up front and assume that it is a numeric. If it isn't, Java will take care of it and throw and exception that you can catch and rethrow or handle.

(You can choose to change this or not. Its up to you.)